File: app/src/main/java/com/qadam/lastpuff/domain/support/MessageCategory.kt
package com.qadam.lastpuff.domain.support

enum class MessageCategory {
    MORNING,
    DAY,
    EVENING,
    SOS_START,
    SOS_15_SEC,
    SOS_30_SEC,
    SOS_45_SEC,
    SOS_60_SEC,
    SOS_75_SEC,
    SOS_90_SEC,
    SOS_105_SEC,
    SOS_120_SEC,
    SOS_140_SEC,
    SOS_LAST_20,
    VICTORY,
    RELAPSE,
    MONEY,
    HEALTH,
    FAMILY,
    CHALLENGE,
    BREATH,
    FACT,
    HUMOR,
    PSYCHOLOGY,
    DIALOG,
    CHECK_IN,
    FUTURE_HOLD,
    WIN_STREAK,
    ACHIEVEMENT_MOMENT
}

enum class SosMode {
    BREATHING,
    FAMILY,
    CHALLENGE;

    companion object {
        fun forDay(dayOfYear: Int): SosMode = entries[dayOfYear % entries.size]
    }
}

data class SupportContext(
    val reasons: List<String> = emptyList(),
    val goalTitle: String? = null,
    val currency: String = "₸",
    val pricePerCigarette: Double = 0.0,
    val moneySaved: Double = 0.0,
    val totalWins: Int = 0,
    val personalLetter: String? = null,
    val hourOfDay: Int = 12
)

data class SosMoment(
    val primary: String,
    val secondary: String? = null,
    val action: String? = null,
    val showPersonalLetter: Boolean = false
)

File: ios/QadamLastPuff/Domain/MessageCategory.swift
import Foundation

enum MessageCategory: String, CaseIterable {
    case morning, day, evening
    case sosStart, sos15Sec, sos30Sec, sos45Sec, sos60Sec, sos75Sec
    case sos90Sec, sos105Sec, sos120Sec, sos140Sec, sosLast20
    case victory, relapse, money, health, family, challenge, breath
    case fact, humor, psychology, dialog, checkIn, futureHold, winStreak, achievementMoment
}

enum SosMode: CaseIterable {
    case breathing, family, challenge

    static func forDay(dayOfYear: Int) -> SosMode {
        allCases[dayOfYear % allCases.count]
    }

    var emoji: String {
        switch self {
        case .breathing: return "🌬"
        case .family: return "❤️"
        case .challenge: return "🎯"
        }
    }
}

struct SupportContext {
    var reasons: [String] = []
    var goalTitle: String?
    var currency: String = "₸"
    var pricePerCigarette: Double = 0
    var moneySaved: Double = 0
    var totalWins: Int = 0
    var personalLetter: String?
    var hourOfDay: Int = 12
}

struct SosMoment {
    let primary: String
    var secondary: String?
    var action: String?
    var showPersonalLetter: Bool = false
}

struct SosComposeFlags {
    var factShown: Bool = false
    var letterShown: Bool = false
}
